spa/src/index.ts

```diff
@@ -58,13 +77,30 @@ function setupApps(modules: Array<[string, System.Module]>) {
 
       if (result && typeof result === "object") {
         System.import(singleSpa).then(({ registerApplication }) => {
-          registerApplication(appName, result.lifecycle, result.activate);
+          registerApplication(
+            appName,
+            result.lifecycle,
+            preprocessActivator(result.activate)
+          );
         });
       }
     }
   }
 }
 
+/**
+ * Runs the shell by importing the translations and starting single SPA.
+ */
+function runShell() {
+  return System.import(singleSpa).then(({ start }) => {
+    return setupLocale()
+      .then(start)
+      .catch((err) => {
+        console.error(`Failed to initialize i18next translations`, err);
```

Comment:
As it's written, this catch block will be hit for an error in either `setupLocale` or `start`, correct?

Reply:
Yes. I guess it would be better to have it before the `start`, i.e.,

```ts
setupLocale().catch(err...).then(start)
```